File: nosh_now_apis/Models/Account.cs
namespace MyApp.Models
{
    public class Account : IModel
    {
        public int Id { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public DateTime CreatedDate { get; set;}
        public int RoleId { get; set; }
        public virtual Role Role {get; set;}
        public virtual Eater Eater{ get; set; }
        public virtual Merchant Merchant{ get; set; }
        public virtual Shipper Shipper{ get; set; }
        public virtual Manager Manager{ get; set; }
    }
}

File: nosh_now_application/lib/data/models/account.dart
import 'package:nosh_now_application/data/models/role.dart';

class Account {
  int accountId;
  String email;
  DateTime createdDate;
  Role? role;
  Account({required this.accountId, required this.email, required this.createdDate, this.role});
  factory Account.fromJson(Map<dynamic, dynamic> json) {
    return Account(
        accountId: json['id'],
        email: json['email'],
        createdDate: DateTime.parse(json['createdDate']),
        role: json['role'] != null ? Role.fromJson(json['role']) : null);
  }
  Map<String, dynamic> toJson() => {
        'id': accountId,
        'email': email,
        'role': role ?? role!.toJson(),
      };
}
